src/front/macOS/kDriveCore/ServerBridge/XPC/XPCConnectionManager.swift

```diff
@@ -0,0 +1,198 @@
+/*
+ Infomaniak kDrive - Desktop
+ Copyright (C) 2023-2025 Infomaniak Network SA
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Foundation
+
+@preconcurrency @objc open class XPCConnectionManager: NSObject, @unchecked Sendable {
+    let machServiceName: String
+
+    var loginItemAgentConnection: NSXPCConnection?
+    var appConnection: NSXPCConnection?
+
+    override init() {
+        guard let loginItemAgentMachName = Bundle.main.object(forInfoDictionaryKey: "LoginItemAgentMachName") as? String else {
+            fatalError("Malformed info.plist, missing LoginItemAgentMachName")
+        }
+
+        IKLogger.xpc.log("[KD] mach name: \(loginItemAgentMachName)")
+        machServiceName = loginItemAgentMachName
+    }
+
+    func scheduleRetryToConnectToLoginAgent() {
+        DispatchQueue.main.async {
+            IKLogger.xpc.log("[KD] Set timer to retry to connect to login agent")
+            Timer.scheduledTimer(withTimeInterval: 10, repeats: false) { [weak self] _ in
+                self?.connectToLoginAgent()
+            }
+        }
+    }
+
+    func connectToLoginAgent() {
+        if loginItemAgentConnection != nil {
+            IKLogger.xpc.log("[KD] Already connected to item agent")
+            return
+        }
+
+        // Initialize connection with login item agent
```

Comment:
You kept the same comments as the existing codebase, but maybe we can remove them now?
Your code is clean and self explanatory, plus the log are helpful to understand what happens.